server/src/main/java/org/opensearch/cluster/metadata/IndexMetadata.java

```diff
@@ -384,6 +385,16 @@ public Iterator<Setting<?>> settings() {
         Property.Dynamic
     );
 
+    /**
+     * Used to specify if the index data should be persisted in the remote store.
+     */
+    public static final Setting<Boolean> INDEX_APPEND_ONLY_ENABLED_SETTING = Setting.boolSetting(
+        SETTING_INDEX_APPEND_ONLY_ENABLED,
+        false,
+        Property.IndexScope,
+        Property.Final
```

Comment:
Should we allow it to be changed from true to false at some later point in the future? I'm ok to have it as final for now but wondering if there are use-cases which require such support.